src/qz/utils/PrefsSearch.java

```diff
@@ -90,6 +92,26 @@ public static int getInt(ArgValue argValue, Properties ... propsArray) {
         return getInt(argValue, true, propsArray);
     }
 
+    public static Integer[] getIntegerArray(ArgValue argValue, Properties ... propsArray) {
+        List<Integer> parsed = parseIntegerArray(getString(argValue, propsArray));
+        return parsed.toArray(new Integer[parsed.size()]);
+    }
+
+    public static List<Integer> parseIntegerArray(String commaSeparated) {
+        List<Integer> parsed = new ArrayList<>();
```

Comment:
Since we know the length this list needs to be, it may be slightly more efficient to pass that size to the array initializer. You would need jungle things around a bit though. If there isn't a clean way to do it, it doesn't really matter. It just prevents the list from over-allocating space.

Reply:
I don't think we know, the list size (generically) can be any number greater than 1 and less than infinity.  Ports are usually in 4s, but the code doesn't mandate that either.  It's rare I use initial size in code as it has a tradeoff of speed versus versatility.  Happy to change my mind if its warranted though.

Reply:
I was thinking more along these lines
```java
    public static List<Integer> parseIntegerArray(String commaSeparated) {
        List<Integer> parsed = null;
        try {
            if (commaSeparated != null && !commaSeparated.isEmpty()) {
                String[] split = commaSeparated.split(",");
                parsed = new ArrayList<>(split.length);
                for(String item : split) {
                    parsed.add(Integer.parseInt(item));
                }
            }
        } catch(NumberFormatException nfe) {
            log.warn("Failed parsing {} as a valid integer array", commaSeparated, nfe);
        }
        if (parsed == null) parsed = new ArrayList<>();
        return parsed;
    }
```
But yea, its a trivial optimization and could be omitted.

Reply:
Right, but if we ever add sanitization in the future, it can still end up being slightly bigger than we need, which the default size of 20 is already doing for most use-cases.  But pragmatically, the purpose of a linked list is to have dynamic sizing, so unless there's a lot resizing happening, the initial size of 20 is generally sane.  If it were a memory-strict language, I'd be more strict, but there's so much GC happening in Java, I've generally left the slightly wasteful default value of 20 to do it's thing. 🤷‍♂️